### PR TITLE
Add OCaml 4.08 and 4.09 in Travis

### PR DESCRIPTION
**Pull request by XVilka**
I guess it's time to check both.

### CODE REVIEW

**Comment by XVilka** [comment]
Strange error:
```
+./configure --prefix /home/travis/local -no-graph -no-debugger -no-ocamldoc -no-ocamlbuild
configure: error: unrecognized option: `-no-graph'
Try `./configure --help' for more information
The command "bash -exu .travis-ci.sh install" failed and exited with 1 during .
```

**Comment by dra27** [comment]
Those options changed name in 4.08

**Comment by AltGr** [comment]
Thanks!

**Comment by XVilka** [comment]
Will update tomorrow, sorry, missed the notification.

**Comment by XVilka** [comment]
Now it is green (failures are Travis own fault as usual).

File: .travis-ci.sh [hunk 1]
```diff
@@ -39,6 +39,11 @@ else
 fi
 set -x
 
+# version comparison helper
+version () {
+	echo "$@" | awk -F. '{ printf("%d%02d%d\n", $1,$2,$3); }'
```
Comment (dra27) [inline .travis-ci.sh]: There’s no need to use `awk` for this - `%{OCAML-VERSION//.}` should do fine

File: .travis-ci.sh [hunk 2]
```diff
@@ -159,16 +164,21 @@ EOF
...
-        cd ocaml-$OCAML_VERSION
+        wget "http://caml.inria.fr/pub/distrib/ocaml-${OCAML_VERSION%.*}/ocaml-$OCAML_VERSION.tar.gz"
+        tar -xzf "ocaml-$OCAML_VERSION.tar.gz"
+        cd "ocaml-$OCAML_VERSION"
+        CONFIGURE_SWITCHES=""
         if [[ $OPAM_TEST -ne 1 ]] ; then
-          CONFIGURE_SWITCHES="-no-ocamldoc"
-          if [[ "$OCAML_VERSION" != "4.02.3" ]] ; then
-            CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES -no-ocamlbuild"
+          if [[ $(version "$OCAML_VERSION") -lt $(version "4.08.0") ]]; then
```
Comment (dra27) [inline .travis-ci.sh]: This can be done without version comparisons: `[ -e configure.ac ]` tells you it’s 4.08+

File: .travis-ci.sh [hunk 3]
```diff
@@ -159,16 +164,21 @@ EOF
...
+            if [[ $(version "$OCAML_VERSION") -gt $(version "4.02.3") ]] ; then
+              CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES -no-ocamlbuild"
+            fi
+            CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES -no-graph -no-debugger -no-ocamldoc"
+          else
+            CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES --disable-debugger --disable-ocamldoc"
           fi
         fi
-        ./configure --prefix ~/local -no-graph -no-debugger ${CONFIGURE_SWITCHES:-}
+        ./configure --prefix ~/local ${CONFIGURE_SWITCHES:-}
```
Comment (dra27) [inline .travis-ci.sh]: For 4.08.0, it would be good to include `--disable-graphics-lib`

**Comment by XVilka** [comment]
Fixed, thanks!

File: .travis-ci.sh [hunk 4]
```diff
@@ -159,16 +159,22 @@ EOF
...
+        wget "http://caml.inria.fr/pub/distrib/ocaml-${OCAML_VERSION%.*}/ocaml-$OCAML_VERSION.tar.gz"
+        tar -xzf "ocaml-$OCAML_VERSION.tar.gz"
+        cd "ocaml-$OCAML_VERSION"
+        CONFIGURE_SWITCHES=""
         if [[ $OPAM_TEST -ne 1 ]] ; then
-          CONFIGURE_SWITCHES="-no-ocamldoc"
-          if [[ "$OCAML_VERSION" != "4.02.3" ]] ; then
-            CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES -no-ocamlbuild"
+          if [[ -e configure.ac ]]; then
+            CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES --disable-debugger --disable-ocamldoc --disable-graphics-libs"
```
Comment (dra27) [inline .travis-ci.sh]:
```suggestion
            CONFIGURE_SWITCHES="--disable-debugger --disable-ocamldoc --disable-debug-runtime
            if [[ ${OCAML_VERSION%.*} = '4.08' ]] ; then
              CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES --disable-graph-lib"
            fi
```

(note that you don't need to quote variables in `[[` - that's the point of it)

File: .travis-ci.sh [hunk 5]
```diff
@@ -159,16 +159,22 @@ EOF
     else
       if [[ ! -x ~/local/bin/ocaml ]] ; then
         echo -en "travis_fold:start:ocaml\r"
-        wget http://caml.inria.fr/pub/distrib/ocaml-${OCAML_VERSION%.*}/ocaml-$OCAML_VERSION.tar.gz
-        tar -xzf ocaml-$OCAML_VERSION.tar.gz
-        cd ocaml-$OCAML_VERSION
+        wget "http://caml.inria.fr/pub/distrib/ocaml-${OCAML_VERSION%.*}/ocaml-$OCAML_VERSION.tar.gz"
+        tar -xzf "ocaml-$OCAML_VERSION.tar.gz"
+        cd "ocaml-$OCAML_VERSION"
+        CONFIGURE_SWITCHES=""
```
Comment (dra27) [inline .travis-ci.sh]:
```suggestion
```
(this initialisation is unnecessary)

File: .travis-ci.sh [hunk 6]
```diff
@@ -159,16 +159,22 @@ EOF
...
-          CONFIGURE_SWITCHES="-no-ocamldoc"
-          if [[ "$OCAML_VERSION" != "4.02.3" ]] ; then
-            CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES -no-ocamlbuild"
+          if [[ -e configure.ac ]]; then
+            CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES --disable-debugger --disable-ocamldoc --disable-graphics-libs"
+          else
+            if [[ "$OCAML_VERSION" != "4.02.3" ]] ; then
+              CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES -no-ocamlbuild"
+            fi
+            CONFIGURE_SWITCHES="$CONFIGURE_SWITCHES -no-graph -no-debugger -no-ocamldoc"
```
Comment (dra27) [inline .travis-ci.sh]:
```suggestion
            CONFIGURE_SWITCHES="-no-graph -no-debugger -no-ocamldoc"
```
(and move it *above* the previous `if`, cf. the other branch)

File: .travis-ci.sh [hunk 7]
```diff
@@ -159,16 +159,22 @@ EOF
...
+
           fi
         fi
-        ./configure --prefix ~/local -no-graph -no-debugger ${CONFIGURE_SWITCHES:-}
+        ./configure --prefix ~/local ${CONFIGURE_SWITCHES:-}
```
Comment (dra27) [inline .travis-ci.sh]:
```suggestion
        ./configure --prefix ~/local $CONFIGURE_SWITCHES
```
(not setting `$CONFIGURE_SWITCHES` is now an error, so it's handy to allow it to propagate)

**Comment by XVilka** [comment]
So it is green and I addressed everything you wrote

**Comment by rjbou** [comment]
Thanks @XVilka & @dra27 !

**Pull request merged**